projects/sbb-esta/angular-public/src/lib/tooltip/tooltip/tooltip.component.ts

```diff
@@ -146,15 +181,17 @@ export class TooltipComponent implements OnDestroy {
 
   private readonly _closeKeyEventStream = new Subject<void>();
   private _closingActionsSubscription: Subscription;
-
-  constructor(
-    private _overlay: Overlay,
-    private _tooltipRegistry: TooltipRegistryService,
-    @Inject(SBB_TOOLTIP_SCROLL_STRATEGY) private _scrollStrategy,
-    @Optional() @Inject(DOCUMENT) private _document: any,
-    private _zone: NgZone,
-    private _changeDetectorRef: ChangeDetectorRef
-  ) {}
+  /**
+   * Customizations for trigger type - delay open/close
+   */
+  @Input() triggerType: 'click' | 'hover' = 'click';
+  @Input() hoverOpenDelay = 0;
+  @Input() hoverCloseDelay = 0;
+  /**
+   * References for timeout used for delay open/close
+   */
+  private _referenceTimoutMouseOver: any;
```

Comment:
Would it maybe simpler to simply track an array of timeout references and clear them as such?
```
private _timeoutReferences: number[] = [];
...
private _clearTimeoutReferences() {
  while (this._timeoutReferences.length) {
    clearTimeout(this._timeoutReferences.shift());
  }
}
```
This is just a thought. What is your opinion?